engine/gamesys/src/gamesys/components/comp_mesh.cpp

```diff
@@ -739,6 +740,86 @@ namespace dmGameSystem
+
+
+        const dmIntersection::Frustum frustum = *params.m_Frustum;
+        uint32_t num_entries = params.m_NumEntries;
+        for (uint32_t i = 0; i < num_entries; ++i)
+        {
+            dmRender::RenderListEntry* entry = &params.m_Entries[i];
+            MeshComponent* component_p = (MeshComponent*)entry->m_UserData;
+
+            dmGameSystem::BufferResource* br = GetVerticesBuffer(component_p, component_p->m_Resource);
+            dmVMath::Vector3 boundsMin, boundsMax;
+
+            void* data;
+            uint32_t count;
+            dmBuffer::ValueType valueType;
+            dmBuffer::Result r = dmBuffer::GetMetaData(br->m_Buffer, dmHashString64("AABB"), &data, &count, &valueType );
+            if (r == dmBuffer::RESULT_METADATA_MISSING) {
+                // no bounding box available - no culling
+                entry->m_Visibility = dmRender::VISIBILITY_FULL; // TODO - or shall we just "continue"
+                continue;
+            } else
+            if (r != dmBuffer::RESULT_OK) {
+                dmLogError("Error getting AABB for mesh buffer");
+                continue;
+            }
+
+            switch (valueType) {
+                case dmBuffer::VALUE_TYPE_FLOAT32:
+                    MetadataAabbToPoints((float*)data, valueType, boundsMin, boundsMax);
+                break;
+                case dmBuffer::VALUE_TYPE_UINT8:
+                    MetadataAabbToPoints((uint8_t*)data, valueType, boundsMin, boundsMax);
+                break;
+                case dmBuffer::VALUE_TYPE_UINT16:
+                    MetadataAabbToPoints((uint16_t*)data, valueType, boundsMin, boundsMax);
+                break;
+                case dmBuffer::VALUE_TYPE_UINT32:
+                    MetadataAabbToPoints((uint32_t*)data, valueType, boundsMin, boundsMax);
+                break;
+                case dmBuffer::VALUE_TYPE_UINT64:
+                    MetadataAabbToPoints((uint64_t*)data, valueType, boundsMin, boundsMax);
+                break;
+                case dmBuffer::VALUE_TYPE_INT8:
+                    MetadataAabbToPoints((int8_t*)data, valueType, boundsMin, boundsMax);
+                break;
+                case dmBuffer::VALUE_TYPE_INT16:
+                    MetadataAabbToPoints((int16_t*)data, valueType, boundsMin, boundsMax);
+                break;
+                case dmBuffer::VALUE_TYPE_INT32:
+                    MetadataAabbToPoints((int32_t*)data, valueType, boundsMin, boundsMax);
+                break;
+                case dmBuffer::VALUE_TYPE_INT64:
+                    MetadataAabbToPoints((int64_t*)data, valueType, boundsMin, boundsMax);
+                break;
+                default:
+                    dmLogError("Invalid typeValue in buffer metadata");
+                    continue;
+
+            }
+
+            bool intersect = dmIntersection::TestFrustumOBB(frustum, component_p->m_World, boundsMin, boundsMax, false);
```

Comment:
Here skip_near_far is hardcoded to 'false'.

I'm not sure how to handle this. When would this be 'true' ?

An idea that came to mind is to "allow" AABB metadata with 4 instead of 6 numbers. If it's 4, then it's 2D culling. Set skip_near_far to 'true'. Otherwise set to 'false'.

Reply:
I don't think we should use 2D aabb.
I think using different functions is better.

Eventually, we probably want the user to set the flag from the render script, just like they do today when choosing the frustum.

Reply:
I'm not sure what you mean.

Shall we use skip_near_far=false for all types of meshes for now in comp_mesh.cpp ?

What about different functions ? Do you mean some sort of overloading in dmIntersection ? Like this ?

```

dmIntersection::TestFrustumOBB2D(frustum, component_p->m_World, boundsMin, boundsMax) {
     dmIntersection::TestFrustumOBB(frustum, component_p->m_World, boundsMin, boundsMax, true);
}

dmIntersection::TestFrustumOBB3D(frustum, component_p->m_World, boundsMin, boundsMax); {
     dmIntersection::TestFrustumOBB(frustum, component_p->m_World, boundsMin, boundsMax, false);
}
```

Reply:
Not entirely sure what I meant with "different functions" here.
I think the current solution works: TestFrustumOBB and AABB with 6 floats.

Reply:
cool. 

No action item then. I'll the conversation open.